dev_guide/builds.adoc

```diff
@@ -519,9 +519,52 @@ customStrategy:
 ----
 ====
 
+[[build-inputs]]
+== Build Inputs
+
+There are several ways to provide content for the build to operate on:  
+
+* Git repositories
+* Inline Dockerfile definitions
+* Binary inputs
+* Content extracted from existing images
+
+These inputs can also be combined in a single build.  If multiple inputs are specified and include overlapping filenames, the inline Dockerfile takes precedence, thereby overwriting any file named Dockerfile provided by another input.  Next in precedence is content from existing images, and finally content from the git repository or binary input.  Binary input and Git repository are mutually exclusive inputs.
```

Comment:
@bparees I feel like a lot of into in the above para could be taken away, and listing the bullet points in order or precedence. So:

"There are several ways to provide content for builds to operate on. In existing order of precedence:

* Inline Dockerfile definitions
* Content extracted from existing images
* Git repositories
* Binary inputs

These can be combined into a single build. As the Dockerfile takes precedence, it can overwrite any other file named Dockerfile provided by another input. Binary input and Git repository are mutually exclusive inputs."

Or something like this.